### PR TITLE
Namespaces by path

### PR DESCRIPTION
**Pull request by teemow**
To use preconfigured network namespaces with nsproxy you can now add the path of the namespace via the flag --namespace (-ns).

eg. nsproxy -ns /var/run/netns/mynamespace -p /mybackends/

### CODE REVIEW

File: nsproxy.go [hunk 1]
```diff
@@ -85,6 +93,7 @@ func main() {
 		cli.StringFlag{"path, p", "", "path to a dir", false},
 		cli.StringFlag{"addr, a", "127.0.0.1:8080", "target address inside the namespace", false},
 		cli.StringFlag{"peers, C", "http://127.0.0.1:4001", "a comma seperated list of machine addresses in the cluster", false},
+		cli.StringFlag{"namespace, ns", "", "target namespace path", false},
```
Comment (philips) [inline nsproxy.go]: What do you think of `-ns-path`? Then we will change the current "target" to `-ns-pid`.

File: nsproxy.go [hunk 2]
```diff
@@ -52,7 +53,14 @@ func proxy(c *cli.Context) {
 
 	go b.Watch(client)
 
-	listener, err := nameNet.ListenNamespace(uintptr(target), "tcp", addr)
+	var listener net.Listener
+	if target != 0 {
```
Comment (philips) [inline nsproxy.go]: You can clean this up a bit by translating target into a namespacePath:

http://godoc.org/github.com/coreos/go-namespaces/namespace#ProcessPath

See the code in nspipe: https://github.com/coreos/go-namespaces/blob/master/nspipe/nspipe.go#L39

**Comment by philips** [comment]
@teemow Thanks a lot! Let me know if you can make the above adjustments- otherwise I can take your code and clean it up.

**Comment by teemow** [comment]
@philips No prob. I did the suggested changes. What do you think?

**Comment by philips** [comment]
looks great, thanks!

**Pull request merged**